### ISSUE TITLE
const assertions is not supported

### ISSUE THREAD

**Issue by JounQin** [gajus/eslint-plugin-jsdoc#846]
https://github.com/microsoft/TypeScript/pull/45464

## Expected behavior

No warning

## Actual behavior

Error

<img width="505" alt="image" src="https://user-images.githubusercontent.com/8336744/156926601-845223af-48b8-4ffc-9e14-1deb3083d9bf.png">

## ESLint Config

```json
{
  "jsdoc/no-undefined-types": "error"
}
```

## ESLint sample

```js
const configs = /** @type {const} */ (['cjs', 'esm'])
```

## Environment

- Node version:  v16.14.0
- ESLint version  v8.10.0
- `eslint-plugin-jsdoc` version: 37.7.1

**Comment by brettz9** [comment]
Thanks for the report. As this relates to type parsing, please file this with [`jsdoc-type-pratt-parser`](https://github.com/jsdoc-type-pratt-parser/jsdoc-type-pratt-parser). Closing as we typically update pretty quickly after its own updates, but please do cross-reference this issue if filing over there. Thanks!

**Comment by JounQin** [comment]
@brettz9 It seems [jsdoc-type-pratt-parser](https://github.com/jsdoc-type-pratt-parser/jsdoc-type-pratt-parser) parsed `{const}` as type correctly, but it is not supported here.

<img width="915" alt="image" src="https://user-images.githubusercontent.com/8336744/156927284-5ca76ac6-b583-4deb-b01d-12f6823c8417.png">

**Comment by brettz9** [comment]
Oh, my apologies, I assumed the error was from `valid-types`. I've added `const` (and also `unknown`).
